ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -484,17 +484,17 @@ let persist_extauth_config ~domain ~user ~ou_conf ~workgroup =
     |> debug "update external_auth_configuration for host %s")
   |> Server_helpers.exec_with_new_task "update external_auth_configuration"
 
-let clean_machine_account ~service_name = function
+let disable_machine_account ~service_name = function
   | Some u, Some p -> (
-      (* Clean machine account in DC *)
+      (* Disable machine account in DC *)
       let env = [|Printf.sprintf "PASSWD=%s" p|] in
-      let args = ["ads"; "leave"; "-U"; u; "-d"; debug_level] in
+      let args = ["ads"; "leave"; "-U"; u; "--keep-account"; "-d"; debug_level] in
```

Comment:
When you leave an AD domain using XenCenter, you can choose whether to leave the machine account enabled or not, so I would have expected to see some boolean switch alongside `--keep-account`?

Reply:
if user name and password is not provides (in XenCenter), then the machine account is not touched, just local leave is performed. 
if user name and password is provided (in XenCenter), we originally  delete the machine account, but we now just disable the machine account like PBIS